File: lingting-spring-ali/src/test/java/live/lingting/spring/ali/SpringAliTest.java
package live.lingting.spring.ali;

import live.lingting.framework.ali.AliOssBucket;
import live.lingting.framework.ali.AliOssObject;
import live.lingting.framework.ali.AliSts;
import live.lingting.framework.aws.AwsS3Bucket;
import live.lingting.framework.aws.s3.interfaces.AwsS3BucketInterface;
import live.lingting.spring.util.SpringUtils;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

/**
 * @author lingting 2024-09-18 20:55
 */
@SpringBootTest
class SpringAliTest {

	@Autowired
	AliSts sts;

	@Autowired
	AliOssBucket ossBucket;

	@Test
	void test() {
		assertNotNull(sts);
		assertNotNull(ossBucket);
		AliOssObject obsObject = ossBucket.use("key");
		assertTrue(obsObject.publicUrl().contains("key"));
		Map<String, AwsS3Bucket> awsS3BucketMap = SpringUtils.getBeansOfType(AwsS3Bucket.class);
		assertEquals(1, awsS3BucketMap.size());
		Map<String, AwsS3BucketInterface> awsS3BucketInterfaceMap = SpringUtils
			.getBeansOfType(AwsS3BucketInterface.class);
		assertEquals(2, awsS3BucketInterfaceMap.size());
	}

}

File: lingting-spring-ali/src/test/kotlin/live/lingting/spring/ali/SpringAliTest.kt
package live.lingting.spring.ali

import live.lingting.framework.ali.AliOssBucket
import live.lingting.framework.ali.AliSts
import live.lingting.framework.aws.AwsS3Bucket
import live.lingting.framework.aws.s3.interfaces.AwsS3BucketInterface
import live.lingting.spring.util.SpringUtils
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

/**
 * @author lingting 2024-09-18 20:55
 */
@SpringBootTest
class SpringAliTest {
    @Autowired
    var sts: AliSts? = null

    @Autowired
    var ossBucket: AliOssBucket? = null

    @Test
    fun test() {
        assertNotNull(sts)
        assertNotNull(ossBucket)
        val obsObject = ossBucket!!.use("key")
        assertTrue(obsObject.publicUrl().contains("key"))
        val awsS3BucketMap = SpringUtils.getBeansOfType(AwsS3Bucket::class.java)
        assertEquals(1, awsS3BucketMap.size)
        val awsS3BucketInterfaceMap = SpringUtils.getBeansOfType(AwsS3BucketInterface::class.java)
        assertEquals(2, awsS3BucketInterfaceMap.size)
    }
}
